### ISSUE TITLE
auto-complete is terrible

### ISSUE THREAD

**Issue by alexezh** [dotnet/vscode-csharp#7592]
Type: <b>Bug</b>

auto-complete is annoying. Empty line, I want to add two tabs. Complete suggests some wrong code. Click esc, tab. It suggests the same again. Another esc..

Extension version: 2.45.25
VS Code version: Code 1.93.0 (Universal) (4849ca9bdf9666755eb463db297b69e5385090e3, 2024-09-04T13:02:38.431Z)
OS version: Darwin arm64 23.6.0
Modes:

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Apple M2 Max (12 x 2400)|
|GPU Status|2d_canvas: enabled<br>canvas_oop_rasterization: enabled_on<br>direct_rendering_display_compositor: disabled_off_ok<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>opengl: enabled_on<br>rasterization: enabled<br>raw_draw: disabled_off_ok<br>skia_graphite: disabled_off<br>video_decode: enabled<br>video_encode: enabled<br>webgl: enabled<br>webgl2: enabled<br>webgpu: enabled<br>webnn: disabled_off|
|Load (avg)|3, 2, 2|
|Memory (System)|64.00GB (3.06GB free)|
|Process Argv|--crash-reporter-id 0a732980-b20c-46f8-b997-5ac8d8ce8f36|
|Screen Reader|no|
|VM|0%|
</details><details>
<summary>A/B Experiments</summary>

```
vsliv368cf:30146710
vspor879:30202332
vspor708:30202333
vspor363:30204092
vscod805cf:30301675
binariesv615:30325510
vsaa593cf:30376535
py29gd2263:31024239
c4g48928:30535728
azure-dev_surveyone:30548225
962ge761:30959799
pythongtdpath:30769146
welcomedialogc:30910334
pythonnoceb:30805159
asynctok:30898717
pythonregdiag2:30936856
pythonmypyd1:30879173
2e7ec940:31000449
pythontbext0:30879054
accentitlementsc:30995553
dsvsc016:30899300
dsvsc017:30899301
dsvsc018:30899302
cppperfnew:31000557
dsvsc020:30976470
pythonait:31006305
dsvsc021:30996838
01bff139:31013167
pythoncenvpt:31062603
a69g1124:31058053
dvdeprecation:31068756
dwnewjupytercf:31046870
2f103344:31071589
impr_priority:31102340
nativerepl1:31104043
refactort:31108082
pythonrstrctxt:31112756
wkspc-onlycs-c:31111717
wkspc-ranged-c:31125598
fje88620:31121564
aajjf12562:31125793

```

</details>

**Comment by dibarbet** [comment]
@alexezh would you be able to provide a gif of the behavior you're seeing?  There could be multiple extensions involved depending on where the auto complete is coming from

**Comment by alexezh** [comment]
I have a small C# file with a function indented at 2 tabs. I have an empty
line in the middle of the class; and I want to type a comment. VS shows
some random suggestion next to IP.  If I type tab, suggestion is accepted
and this is not what I want. So I click Esc, Tab. VS idents IP and
immediately shows the same suggestion. I have to recognize this and do
Esc/Tab again.

Is there a way to disable suggestions until I actually start typing? The
current complete logic is not good at guessing without input, so I would
rather not spend time clicking Esc over and over.

**Comment by dibarbet** [comment]
@alexezh generally completion from the C# extension shouldn't be active until you start typing.  I'm wondering if there is some other extension providing completions.  Would you mind sharing either a gif or a screenshot of the suggestion?  And what extensions you have installed?

**Comment by oakad** [comment]
It happens all the time now.

Here I moved to new line. If I were to press `<tab>` the return statement will get inserted into the buffer. I can press `<esc>` to make it go, but it immediately appears again (see below).
Nothing in the codebase should imply that I ever wanted to return null, of all things. :-)

![Image](https://github.com/user-attachments/assets/97b3b743-f66f-4c14-91da-c428813095da)
![Image](https://github.com/user-attachments/assets/ba544fe8-cb69-4d55-a23c-e25c0e2e9b2c)

**Comment by oakad** [comment]
The entire issue is coming from "IntelliCode for C# Dev Kit". It is really bad, making the editor totally unusable - every attempt at indentation with tabs spills out some bogus suggestion.

Below is a small excerpt from the log - suddenly, the escape is my most pressed button.

> [WLC] [CompletionProvider] (#7687) Recieved cancellation event, ran for 124.0084171295166
[WLC] [CompletionProvider] (#7688) Recieved cancellation event, ran for 111.14845895767212
[WLC] [CompletionProvider] (#7689) Current line with candidate:
						sortChanged = true;
[WLC] [CompletionProvider] (#7691) Current line with candidate:

> [WLC] [CompletionProvider] (#7692) Current line with candidate:
						break;
[WLC] [CompletionProvider] (#7693) Recieved cancellation event, ran for 74.84245800971985

**Comment by oakad** [comment]
It was kind of tricky to locate the correct GitHub project for intellicode extension. :-)

**Comment by dibarbet** [comment]
> It was kind of tricky to locate the correct GitHub project for intellicode extension. :-)

Agreed 😆 - easiest way I've found is the issues link on the [marketplace page](https://marketplace.visualstudio.com/items?itemName=ms-dotnettools.vscodeintellicode-csharp)
